### PR TITLE
Share clink

### PR DESCRIPTION
**Pull request by alisa-lin**
## Changelog
- fetch all users and fetch users with write access. compute users without write access on the redux layer
- populate share modal with corresponding users list from redux state
- add unshare ability and separate share and unshare so they appear in the same modal but are different forms
- hook them up so they correctly write to the read & write maps

## Screenshots
Screenshot 2020-07-29 at 5 35 04 PM
Screenshot 2020-07-29 at 5 35 16 PM

### CODE REVIEW

File: react-app/src/components/Common/clinkmenu.js [hunk 1]
```diff
@@ -136,32 +157,60 @@ const ClinkMenu = (props) => {
...
             ]}
           >
-            {/* clink url goes here */}
+            <Select mode="multiple" 
+              filterOption={(input, option) =>
+                option.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
+              }>
+              {sharedUsers.map(user => (
+                <Option value={user.id}>{user.username}</Option>
+              ))}
```
Comment (clairezc27) [inline react-app/src/components/Common/clinkmenu.js:203]: should we exclude ourselves as an option to remove write access from? if not, when I remove write access for myself on a clink that I own, will the clink not display in the sidebar/user page?

File: react-app/package.json [hunk 2]
```diff
@@ -63,7 +63,6 @@
     "style-loader": "1.0.0",
     "terser-webpack-plugin": "1.4.1",
     "ts-pnp": "1.1.4",
-    "underscore": "^1.10.2",
     "url-loader": "2.1.0",
     "webpack": "4.41.0",
     "webpack-dev-server": "3.2.1",
```
[the remark this answers is not in the document]
Comment (alisa-lin) [reply react-app/package.json:68]: package.json should be tracked

File: react-app/src/components/Common/clinkmenu.js [hunk 3]
```diff
@@ -136,32 +157,60 @@ const ClinkMenu = (props) => {
...
+            </Select>
+          </Form.Item>
+          <Form.Item>
+            <Button type="primary" htmlType="submit" loading={isUnshareLoading}>
+              Submit
+            </Button>
```
[the remark this answers is not in the document]
Comment (alisa-lin) [reply react-app/src/components/Common/clinkmenu.js:209]: I don't think that UI is super clear.

File: react-app/src/components/Common/clinkmenu.js (hunk 1 above)
Comment (alisa-lin) [reply react-app/src/components/Common/clinkmenu.js:203]: Yeah I was wondering. but if you were trying to remove yourself, how would you do it?
UX/UI wise there is probably a better way to do this, but we're a week out from launch so I'd leave it and try to get as much of the other functionality up as possible.
Unless we don't want the ability to remove yourself at all? Only other people can remove you?

File: react-app/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (marsw42) [reply react-app/package.json:68]: This package might be used by Michael on search, could you confirm with him?

File: react-app/src/components/Common/clinkmenu.js [hunk 4]
```diff
@@ -136,32 +157,60 @@ const ClinkMenu = (props) => {
...
           </Form.Item>
         </Form>
+          Share read access by link:
+            {/* clink url goes here */}
```
[the remark this answers is not in the document]
Comment (alisa-lin) [reply react-app/src/components/Common/clinkmenu.js:211]: I wasn't sure what to put... because right now all our URLs are localhost

**Pull request merged**